MathGamev1

RyanW84 opened the pull request:
My first version of Math Game for your review

RyanW84 committed:
MathGamev1

My first version of Math Game for your review

RyanW84 committed:
fixed codacy issues

fixed codacy issues

RyanW84 committed:
fixed codacy issues 2

fixed codacy issues 2

TheCSharpAcademy commented:
@RyanW84 Hey there! How's it going? Just wanted to let you know that if you want to update the project you can just commit and push your code and I'll be notified. Looking forward to our meeting!

TheCSharpAcademy reviewed (changes requested):
@RyanW84

RyanW84 committed:
refactoring

really stuck with enums and getting them to work

RyanW84 committed:
Really stuck

stuck with Enums and refactoring

RyanW84 committed:
commit again

stuck with enums and classes and methods and refactoring, helP!

RyanW84 committed:
stuck

stuck with enums etc

RyanW84 committed:
Enum issue fixed

Enum issue fixed and mostly refactored

RyanW84 committed:
Codacy improvements made

Removed unneccessary commented out code.

tidied up usings

set most variables to null rather than 0 or false

RyanW84 committed:
I believe I have fixed the following issues:
Division by Zero eliminated
Tuple swap for X and Y so that the larger number is always presented first in an addition
Menu Bugs fixed
Random mode added
ScoreBoard addedCodacy issues fixed

Tidied up usings

Tidied up setting variables to 0 unnecessarily

removed large chunks of commented out code

RyanW84 committed:
Codacy issues fixed

Tidied up usings

Tidied up setting variables to 0 unnecessarily

removed large chunks of commented out code

TheCSharpAcademy reviewed (changes requested):
@RyanW84 Looks like your code compile and the difficulty works fine! Can you please fix these before I approve?

1 - You're still showing divisions that don't result in a whole number:
![image](https://github.com/user-attachments/assets/accbf0ba-db69-4dd2-ba50-8f21e88ac0b9)

2 - All other operations now only show 0s: 
![image](https://github.com/user-attachments/assets/fa9cfaf2-45f2-478d-93f7-8d9280df63d7)

RyanW84 commented:
Hi,

Yes i noticed these issues after our meeting.

They are on my to do list as I want to get the Math Game as complete as
possible.

I believe the sums displaying 0 issues is a selection criteria issue.


Regarding  division, I have implemented a tuple swap when the code detects
Y is bigger than x.

I will give it all a thorough look over  and send another commit once I've
fixed these issues.

Thank you for highlighting these issues

On Sat, 8 Feb 2025, 00:13 The C# Academy, ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> @RyanW84 <https://github.com/RyanW84> Looks like your code compile and
> the difficulty works fine! Can you please fix these before I approve?
>
> 1 - You're still showing divisions that don't result in a whole number:
> image.png (view on web)
> <https://github.com/user-attachments/assets/accbf0ba-db69-4dd2-ba50-8f21e88ac0b9>
>
> 2 - All other operations now only show 0s:
> image.png (view on web)
> <https://github.com/user-attachments/assets/fa9cfaf2-45f2-478d-93f7-8d9280df63d7>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/469#pullrequestreview-2603107912>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKEDBVTPHDQ2V37KZOEES5L2OVD2HAVCNFSM6AAAAABWBP4ZEKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMMBTGEYDOOJRGI>
> .
> You are receiving this because you were mentioned.Message ID:
> <TheCSharpAcademy/CodeReviews.Console.MathGame/pull/469/review/2603107912@
> github.com>
>

RyanW84 committed:
Removed unnecessary using Merge branch 'main' of https://github.com/RyanW84/CodeReviews.Console.MathGame

RyanW84 committed:
Merge branch 'main' of https://github.com/RyanW84/CodeReviews.Console.MathGame

RyanW84 committed:
I believe I have fixed the following issues:
Division by Zero eliminated
Tuple swap for X and Y so that the larger number is always presented first in an addition
Menu Bugs fixed
Random mode added
ScoreBoard addedCodacy issues fixed

Tidied up usings

Tidied up setting variables to 0 unnecessarilyMerge branch 'main' of https://github.com/RyanW84/CodeReviews.Console.MathGame

RyanW84 committed:
Merge branch 'main' of https://github.com/RyanW84/CodeReviews.Console.MathGame

RyanW84 committed:
Merge branch 'main' of https://github.com/RyanW84/CodeReviews.Console.MathGame

RyanW84 committed:
I believe I have fixed the following issues:
Division by Zero eliminated
Tuple swap for X and Y so that the larger number is always presented first in an addition
Menu Bugs fixed
Random mode added
ScoreBoard addedCodacy issues fixed

Tidied up usings

Tidied up setting variables to 0 unnecessarily

RyanW84 committed:
I believe I have fixed the following issues:
Division by Zero eliminated
Tuple swap for X and Y so that the larger number is always presented first in an addition
Menu Bugs fixed
Random mode added
ScoreBoard addedCodacy issues fixed

Tidied up usings

Tidied up setting variables to 0 unnecessarily

RyanW84 committed:
Fixes applied

I believe I have fixed the following issues:
Division by Zero eliminated
Tuple swap for X and Y so that the larger number is always presented first in an addition
Menu Bugs fixed
Random mode added
ScoreBoard addedCodacy issues fixed

Tidied up usings

Tidied up setting variables to 0 unnecessarily

RyanW84 committed:
Create README.md

RyanW84 committed:
Update README.md

TheCSharpAcademy commented:
@RyanW84   Project approved! 😄✅ And well done for completing the challenges! I recommend you always do that, as it pushes you and makes you learn faster. 💪🏻

There was still one requirement missing, you were still showing divisions like 6/5. Maybe you didn't push your final solution? Here's how you'd solve it: You'd use the % for your operation ,which gives you the remainder. If it's anything other than 0, it's an invalid question.
![image](https://github.com/user-attachments/assets/8dc0458d-bab6-4f43-bd92-7851121fed8d).

I approved anyways since you proved to have a good understanding of C# foundations, which is the purpose of this project. Plus I'd like you to move on to SQL quick 😎

Looking forward to seeing your Habit Tracker!

RyanW84 commented:
Thank you,

I've had a look and yep accidentally missed off the modulus after
refactoring the Game logic. Thanks for pointing it out.

I will add this to my code later, as I want to have it 100% when it is in
my repository in case a potential employer looks through my GitHub at some
point.

I'm really enjoying the course and learning so much and at a much faster
pace than I expected.

I've started on the habit tracker and created my first  database and added
values to it through C# 🙌

On Tue, 11 Feb 2025, 06:26 The C# Academy, ***@***.***> wrote:

> @RyanW84 <https://github.com/RyanW84> Project approved! 😄✅ And well done
> for completing the challenges! I recommend you always do that, as it pushes
> you and makes you learn faster. 💪🏻
>
> There was still one requirement missing, you were still showing divisions
> like 6/5. Maybe you didn't push your final solution? Here's how you'd solve
> it: You'd use the % for your operation ,which gives you the remainder. If
> it's anything other than 0, it's an invalid question.
> image.png (view on web)
> <https://github.com/user-attachments/assets/8dc0458d-bab6-4f43-bd92-7851121fed8d>
> .
>
> I approved anyways since you proved to have a good understanding of C#
> foundations, which is the purpose of this project. Plus I'd like you to
> move on to SQL quick 😎
>
> Looking forward to seeing your Habit Tracker!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/469#issuecomment-2649903198>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKEDBVTP7UKGULZV6R2Q7AT2PGJZ7AVCNFSM6AAAAABWBP4ZEKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNBZHEYDGMJZHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> .com>
>
